devito/ir/support/space.py

```diff
@@ -8,11 +8,27 @@
 
 from devito.ir.support.vector import Vector, vmin, vmax
 from devito.tools import (PartialOrderTuple, as_list, as_tuple, filter_ordered,
-                          frozendict, toposort, is_integer)
+                          frozendict, is_integer, toposort)
 from devito.types import Dimension, ModuloDimension
 
-__all__ = ['NullInterval', 'Interval', 'IntervalGroup', 'IterationSpace', 'DataSpace',
-           'Forward', 'Backward', 'Any']
+__all__ = ['Stamp', 'NullInterval', 'Interval', 'IntervalGroup', 'IterationSpace',
+           'DataSpace', 'Forward', 'Backward', 'Any']
+
+
+class Stamp(object):
+
+    """
+    Uniquely identifies Intervals.
```

Comment:
Not so sure actually, this is a Stamp for Intervals, I wouldn't want to have different classes of objects using the same kind of stamps?

Reply:
I don't think it would be a problem, I don't see why a stamp would be compared between different classes. Just thought it may be better than having to create a new Stanmp class anytime we'll need a new stamping thing